initramfs/scripts/bootloader-script

```diff
@@ -0,0 +1,190 @@
+# -*- shell-script -*- vim:ft=sh:
+#---------------------------------------------------------------------
+# Description: Update kernel/core snap.
+# Entry point: mountroot().
+#---------------------------------------------------------------------
+
+pre_mountroot()
+{
+	local script_dir="/scripts/local-top"
+	log_begin_msg "Running $script_dir"
+	run_scripts "$script_dir"
+	log_end_msg
+}
+
+# Determine full path to disk partition given a filesystem label.
+get_partition_from_label()
+{
+	local label="$1"
+
+	[ -n "$label" ] || panic "need FS label"
+
+	# Make sure the device has been created by udev before looking for it
+	# Don't need to panic, since the output will be validated outside this function
+	wait-for-root "LABEL=$label" "${ROOTDELAY:-180}" >/dev/null || true
+
+	local part=$(find /dev -name "$label"|tail -1)
+	[ -z "$part" ] && return
+	local path=$(readlink -f "$part")
+	[ -n "$path" ] && echo "$path"
+}
+
+# setup $rootmnt based on os/kernel snaps
+do_root_mounting()
```

Comment:
Same for this one,  I wonder if we can share code with ubuntu-core-rootfs here. Probably needs a small refactor because the ubuntu-c-r is doing slightly more.